### PR TITLE
Selective Database Restore

### PR DESCRIPTION
**Pull request by dwsteele**
This feature allows only selected databases to be restored. Databases not specifically included will be restored as sparse, zeroed files to save space but still allow PostgreSQL to perform recovery. After recovery the databases that were not included will not be accessible but can be remove with the drop database command.

Note that built-in databases (template0, template1, and postgres) are always restored.

### CODE REVIEW

File: doc/xml/user-guide.xml [hunk 1]
```diff
@@ -740,6 +740,154 @@
...
             </execute-list>
         </section>
+
+        <!-- SECTION => RESTORE - DELTA -->
+        <section id="option-db-include">
+            <title>Restore Selected Databases</title>
+
+            <p>There may be cases where it is desirable to selectively restore specific databases from a cluster backup.  This could be done for performance reasons or to move selected databases to a machine that does not have enough space to restore the entire cluster backup.</p>
+
+            <p>To demonstrate this feature two databases are be created: test1 and test2.  A new backup is run so <backrest/> is aware of the new databases.</p>
```
Comment (cmwshang) [inline doc/xml/user-guide.xml]: typo: To demonstrate this feature two databases are TO be created (or just say "are created")

Comment (dwsteele) [inline doc/xml/user-guide.xml]: Thanks, fixed!

File: lib/pgBackRest/Restore.pm [hunk 2]
```diff
@@ -1082,6 +1082,84 @@ sub process
...
+            {
+                confess &log(ERROR, "system databases (template0, postgres, etc.) are included by default", ERROR_DB_INVALID);
+            }
+
+            # Otherwise remove from list of DBs to zero
+            delete($oDbList{$strDbKey});
+        }
+
+        # Construct regexp for identify files that should be zeroed
+        for my $strDbKey (sort(keys(%oDbList)))
```
Comment (cmwshang) [inline lib/pgBackRest/Restore.pm]: comment typo: Construct regex TO identify.

Comment (dwsteele) [inline lib/pgBackRest/Restore.pm]: @cmwshang Fixed.

File: doc/xml/reference.xml [hunk 3]
```diff
@@ -334,6 +334,19 @@
...
+                    <!-- CONFIG - RESTORE SECTION - DB-INCLUDE KEY -->
+                    <config-key id="db-include" name="Include Database">
+                        <summary>Restore only specified databases.</summary>
+
+                        <text>This feature allows only selected databases to be restored.  Databases not specifically included will be restored as sparse, zeroed files to save space but still allow <postgres/> to perform recovery.  After recovery the databases that were not included will not be accessible but can be remove with the <id>drop database</id> command.
+
+                        Note that built-in databases (<id>template0</id>, <id>template1</id>, and <id>postgres</id>) are always restored.
+
+                        The <br-setting>{[dash]}-db-include</br-setting> option can be password multiple times to specify more than one database to include.</text>
+
```
Comment (cmwshang) [inline doc/xml/reference.xml]: typos: 
line 341: but can be removeD  
line 345: change password to passed

Comment (dwsteele) [inline doc/xml/reference.xml]: @cmwshang Fixed.

**Comment by dwsteele** [comment]
Committed in 0c320e7df.

**Pull request closed without merging**